src/components/SendNativeToken/SendNativeTokenUI.tsx

```diff
@@ -0,0 +1,178 @@
+  onlyEstimate = false,
+}: SendNativeTokenUIProps): React.ReactElement => {
+
+  // State for the amount input
+  const [amount, setAmount] = useState('0.001');
+  // State for checking if the receiver address is valid
+  const [isValidAddress, setIsValidAddress] = useState(true);
+  // State for handling errors
+  const [error, setError] = useState('');
+
+  // Hooks from the EtherspotTransactionKit for transaction estimation and sending
+  const { estimate, send } = useEtherspotTransactions();
+
+  // Effect to add and remove event listener for triggerElement click
+  useEffect(() => {
+    const triggerElementRef = triggerElement as React.MutableRefObject<HTMLElement> | undefined;
+    if (triggerElementRef && triggerElementRef.current) {
+      triggerElementRef.current.addEventListener('click', estimateAndSend);
+    }
+
+    // Cleanup
+    return () => {
+      if (triggerElementRef && triggerElementRef.current) {
+        triggerElementRef.current.removeEventListener('click', estimateAndSend);
+      }
+    };
+  }, [triggerElement]);
+
+  // Effect to validate the receiver address
+  useEffect(() => {
+    if (isValidEthereumAddress(receiverAddress)) {
+      setIsValidAddress(true);
+    } else {
+      setIsValidAddress(false);
+    }
+  }, [receiverAddress]);
+
+  // Handling key press events, specifically Enter key, for sending transactions
+  const handleEnterPress = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (!disableSendOnEnter && event.key === 'Enter') {
+      estimateAndSend();
+    }
+  };
+
+  // Handling input change events for the amount
+  const handleAmountChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setAmount(event.target.value);
+  };
+
+  // Logic for estimating and sending transactions
+  const estimateAndSend = async () => {
+    try {
+      if (isValidAddress) {
+        // Estimate logic
+        const gasEstimation = await estimate();
+  
+        if (!onlyEstimate) {
+          // Send logic
+          await send();
+          console.log('Transaction sent successfully');
+        }
+        // Log gas estimation result
+        console.log('Gas estimation result:', gasEstimation);
+      } else {
+        setError('Receiver address is not a valid blockchain address. Please double-check and try again.');
+      }
+    } catch (error) {
+      if (debug) {
+        console.error('Error in estimateAndSend:', error);
+  
+        if (error instanceof SomeSpecificEstimationError) {
+          console.error('Specific error in gas estimation:', error);
+        } else if (error instanceof SomeSpecificSendError) {
+          console.error('Specific error in sending transaction:', error);
+        }
+      }
+  
+      setError(
+        `An error occurred during the transaction. Please try again.${
+          debug ? ' Check the console for more details.' : ''
+        } If the issue persists, refer to the Etherspot documentation: https://etherspot.dev/docs`
```

Comment:
This part of the message "If the issue persists, refer to the Etherspot documentation: https://etherspot.dev/docs" should be moved to inside the debug === true clause, as this is a developer focused message not a user message